gbus/message.go

```diff
@@ -12,9 +15,36 @@ type BusMessage struct {
 	Payload           interface{}
 }
 
-//NewBusMessage factory mehtod for creating a BusMessage that wraps the given payload
-func NewBusMessage(payload interface{}) BusMessage {
-	return BusMessage{
+//NewBusMessage factory method for creating a BusMessage that wraps the given payload
+func NewBusMessage(payload interface{}) *BusMessage {
+	return &BusMessage{
 		ID:      xid.New().String(),
-		Payload: payload}
+		Payload: payload,
+	}
+}
+
+func NewFromAMQPHeaders(headers amqp.Table) *BusMessage {
```

Comment:
Would prefer the translation between amqp headers and busmessages will be placed in bus.go